src/Tutorial/ui/DocumentationRoot.tsx

```diff
@@ -0,0 +1,54 @@
+import React, { useEffect } from "react";
+
+import Button from "@mui/material/Button";
+import { MD } from "../../ui/md/md";
+
+import { getPage } from "./root";
+import { Navigator, useHistory } from "../../ui/React/Documentation";
+
+const resolveRelativePath = (folder: string, relative: string): string => {
+  const noLastSlash = folder.endsWith("/") ? folder.slice(0, folder.length - 1) : folder;
+  const lastIndex = noLastSlash.lastIndexOf("/");
+  const prefix = lastIndex === -1 ? "" : noLastSlash.slice(0, lastIndex + 1);
+  const suffix = relative.slice("../".length);
+  return prefix + suffix;
+};
+
+const resolvePath = (currentPath: string, newPath: string): string => {
+  const lastIndex = currentPath.lastIndexOf("/");
+  const folder = lastIndex === -1 ? "" : currentPath.slice(0, lastIndex + 1);
+  if (!newPath.startsWith("../")) return folder + newPath;
+
+  return resolveRelativePath(folder, newPath);
+};
+
+export function DocumentationRoot(): React.ReactElement {
+  const history = useHistory();
+  const page = getPage(history.page);
+  const navigator = {
+    navigate(relPath: string, external: boolean) {
+      const newPath = resolvePath(history.page, relPath);
```

Comment:
Use existing pathing helpers, and add a console.error for debugging if somehow we get an unresolvable path (e.g. ../ above top level or an invalid character)
```suggestion
      // resolveFilePath uses absolute path unless relative part starts with ./ or ../
      relPath = "./" + relPath;
      const newPath = isFilePath(relPath) && resolveFilePath(relPath, history.page);
      if (!newPath) {
        console.error(`${relPath} was not a valid filepath while navigating documentation from ${history.page}`);
        return;
      }
```